From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 20 Nov 2018 23:01:46 +0000
Subject: Revert "ALSA: core: Assure control device to be registered at last"
Forwarded: not-needed

This reverts commit 89bfe6fdb301a414d52bbcf93ac0159bd4c5adf6, which
was commit dc82e52492f684dcd5ed9e4773e72dbf2203d75e upstream.  It
causes an ABI change and it's not clear that it's actually important.
---
 include/sound/core.h | 2 +-
 sound/core/device.c  | 9 ---------
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/include/sound/core.h b/include/sound/core.h
index 7436de7b6983..1df3f2fe5350 100644
--- a/include/sound/core.h
+++ b/include/sound/core.h
@@ -51,6 +51,7 @@ struct completion;
  */
 enum snd_device_type {
 	SNDRV_DEV_LOWLEVEL,
+	SNDRV_DEV_CONTROL,
 	SNDRV_DEV_INFO,
 	SNDRV_DEV_BUS,
 	SNDRV_DEV_CODEC,
@@ -61,7 +62,6 @@ enum snd_device_type {
 	SNDRV_DEV_SEQUENCER,
 	SNDRV_DEV_HWDEP,
 	SNDRV_DEV_JACK,
-	SNDRV_DEV_CONTROL,	/* NOTE: this must be the last one */
 };
 
 enum snd_device_state {
diff --git a/sound/core/device.c b/sound/core/device.c
index bcadadeebb19..41bec3075ae5 100644
--- a/sound/core/device.c
+++ b/sound/core/device.c
@@ -219,15 +219,6 @@ void snd_device_free_all(struct snd_card *card)
 
 	if (snd_BUG_ON(!card))
 		return;
-	list_for_each_entry_safe_reverse(dev, next, &card->devices, list) {
-		/* exception: free ctl and lowlevel stuff later */
-		if (dev->type == SNDRV_DEV_CONTROL ||
-		    dev->type == SNDRV_DEV_LOWLEVEL)
-			continue;
-		__snd_device_free(dev);
-	}
-
-	/* free all */
 	list_for_each_entry_safe_reverse(dev, next, &card->devices, list)
 		__snd_device_free(dev);
 }
